checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -426,39 +428,39 @@ private void trackInvocationResult(Set<Obligation> obligations, Node node) {
       return;
     }
 
-    // `mustCallAlias` is the argument passed in the MustCallAlias position, if any exists,
-    // otherwise null.
-    Node mustCallAlias = getMustCallAliasArgumentNode(node);
-    // If mustCallAlias is null and call returns @This, set mustCallAlias to the receiver.
-    if (mustCallAlias == null
-        && node instanceof MethodInvocationNode
+    // mustCallAliases is a (possibly-empty) list of arguments passed in a MustCallAlias position
+    List<Node> mustCallAliases = getMustCallAliasArgumentNodes(node);
+    // If call returns @This, add the receiver to mustCallAliases.
+    if (node instanceof MethodInvocationNode
         && typeFactory.returnsThis((MethodInvocationTree) tree)) {
-      mustCallAlias =
-          removeCastsAndGetTmpVarIfPresent(((MethodInvocationNode) node).getTarget().getReceiver());
+      mustCallAliases.add(
+          removeCastsAndGetTmpVarIfPresent(
+              ((MethodInvocationNode) node).getTarget().getReceiver()));
     }
 
     ResourceAlias tmpVarAsResourceAlias = new ResourceAlias(new LocalVariable(tmpVar), tree);
-    if (mustCallAlias instanceof FieldAccessNode) {
-      // Do not track the call result if the MustCallAlias argument is a field.  Handling of
-      // @Owning fields is a completely separate check, and there is never a need to track an alias
-      // of
-      // a non-@Owning field, as by definition such a field does not have obligations!
-    } else if (mustCallAlias instanceof LocalVariableNode) {
-      Obligation obligationContainingMustCallAlias =
-          getObligationForVar(obligations, (LocalVariableNode) mustCallAlias);
-      // If mustCallAlias is a local variable already being tracked, add tmpVarAsResourceAlias
-      // to the set containing mustCallAlias.
-      if (obligationContainingMustCallAlias != null) {
-        Set<ResourceAlias> newResourceAliasSet =
-            FluentIterable.from(obligationContainingMustCallAlias.resourceAliases)
-                .append(tmpVarAsResourceAlias)
-                .toSet();
-        obligations.remove(obligationContainingMustCallAlias);
-        obligations.add(new Obligation(newResourceAliasSet));
+    for (Node mustCallAlias : mustCallAliases) {
+      if (mustCallAlias instanceof FieldAccessNode) {
+        // Do not track the call result if the MustCallAlias argument is a field.  Handling of
+        // @Owning fields is a completely separate check, and there is never a need to track an
+        // alias of a non-@Owning field, as by definition such a field does not have obligations!
+      } else if (mustCallAlias instanceof LocalVariableNode) {
```

Comment:
Rather than doing `else if` here, just do an `else`, and throw a `TypeSystemError` at the beginning of the `else` block if `!(mustCallAlias instanceof LocalVariableNode)`.

Reply:
https://github.com/typetools/checker-framework/pull/4808/commits/7de173ecf89a3877e76f3e2e5502dbd3c07e1e7c